zerver/models.py

```diff
@@ -900,18 +900,22 @@ def emails_from_ids(user_ids: Sequence[int]) -> Dict[int, str]:
         rows = UserProfile.objects.filter(id__in=user_ids).values('id', 'email')
         return {row['id']: row['email'] for row in rows}
 
-    def can_create_streams(self) -> bool:
-        diff = (timezone_now() - self.date_joined).days
+    def is_allowed_waiting_period(self) -> bool:
         if self.is_realm_admin:
             return True
-        if self.realm.create_stream_by_admins_only:
-            return False
         if self.is_guest:
             return False
+
+        diff = (timezone_now() - self.date_joined).days
         if diff >= self.realm.waiting_period_threshold:
             return True
         return False
 
+    def can_create_streams(self) -> bool:
+        if self.realm.create_stream_by_admins_only:
+            return False
```

Comment:
This part of the commit is wrong; we need to check if the user is an admin before we check this.

Reply:
@timabbott Made the required changes.